Frameworks/AddressBook/ABSourceInternal.mm

```diff
@@ -0,0 +1,31 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import "ABSourceInternal.h"
+#import "UWP/WindowsApplicationModelContacts.h"
+
+@implementation _ABSource
+
+- (id)initWithRecordID:(ABRecordID)recordID {
+    self = [super init];
+    if (self) {
+        self.recordID = recordID;
```

Comment:
>self.recordID = recordID; [](start = 8, length = 25)

I'm assuming that the reason why we want this initializer, rather than just changing the field in ABPersonCopySource, is that we want recordID to be immutable. If this is the case, you ought to use self->_recordID here, as self.recordID is compiled to call the setter that you don't want to have.